algebird-core/src/main/scala/com/twitter/algebird/ExpHist.scala

```diff
@@ -0,0 +1,199 @@
+package com.twitter.algebird
+
+/**
+ * Exponential Histogram algorithm from
+ * http://ilpubs.stanford.edu:8090/504/1/2001-34.pdf.
+ *
+ *
+ * Next step - code up the l-canonical representation of numbers,
+ * then do some of the fancier business in the paper!
+ */
+object ExpHist {
+  case class Config(k: Int, windowSize: Long) {
+    // Maximum number of buckets of size 2^i allowed in the repr of
+    // this exponential histogram.
+    def maxBuckets: Int = math.ceil(k / 2.0).toInt + 1
+
+    // Returns the last timestamp before the window. any ts <= [the
+    // returned timestamp] is outside the window.
+    def expiration(currTime: Long): Long = currTime - windowSize
+  }
+
+  def empty(k: Int, windowSize: Long): ExpHist =
+    ExpHist(Config(k, windowSize), Vector.empty, 0L)
+}
+
+case class ExpHist(conf: ExpHist.Config, buckets: Vector[BucketSeq], time: Long) {
+  /**
+   * Returns the same ExpHist with a new window. If the new window is
+   * smaller than the current window, evicts older items.
+   */
+  def withWindow(newWindow: Long): ExpHist =
```

Comment:
I'm not using this yet - just wanted to note that a key part of the algo is that the window size doesn't really matter, and it's easy to rescale.